### PR TITLE
Dblatcher/ar newsletter signup layout component form

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?

No customer-facing changes - updates to WIP `NewsletterSignupLayout`.

 - Add the `InPageNewsletterSignup` component (with story)
 - The NewsletterSignUpLayout uses the `InPageNewsletterSignup` for the main content, instead of the `ArticleBody`, but uses the `ArticleBody` as a fallback if the client app does not support sign-up components (requires Bridget 2.0.0 -  see https://github.com/guardian/dotcom-rendering/pull/6464).
 - The `initSignUpForms` function will display the fallback content when sign-up components are not supported, as well as removing the form containers 
 - includes the "Newsletter frequency" in the Layout.
 - Moves the `EmailSignupForm` to its own folder as it is now used by both the `InPageNewsletterSignup` and `NewsletterSignup`(in-article) components.

## Why?
Further incremental changes to the WIP layout started in https://github.com/guardian/dotcom-rendering/pull/6223

Provide "Backward compatibility" for App users not upgraded to Bridget 2.0.0 versions at the time the Layout is released.

## Screenshots

| Before      | After (Bridget v1)  | After (Bridget v2) 
|-------------|------------|------------|
| Screenshot 2022-11-30 at 15 59 25| Screenshot 2022-11-30 at 15 59 39| Screenshot 2022-11-30 at 16 01 57

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

**Review by georgeblahblah — COMMENTED** [review]
Looks good!

I had a question around the reasoning behind `initiallyRender` in `InPageNewsletterSignup`. I think we could omit this prop and show the form if Bridget is the correct version, otherwise the fallback/error but I'm interested to hear your thoughts - I'm probably missing some context.

**Comment by dblatcher** [comment]
> Looks good!
> 
> I had a question around the reasoning behind `initiallyRender` in `InPageNewsletterSignup`. I think we could omit this prop and show the form if Bridget is the correct version, otherwise the fallback/error but I'm interested to hear your thoughts - I'm probably missing some context.

My thinking is that rendering the 'waiting' content (or nothing) until after the article script has checked the Bridget version and updated the DOM to reveal either the form or the  fallback content isn't ideal for performance, especially as its the main part of the page - even if the it doesn't take long, updating the DOM clientside after an async check can't be great.

If we were to release this Layout when no users outside Beta have the Bridget 2+ it would seem better to not do the check when the answer will always be 'no'.

I may be overthinking, though -  should it be decided  go live with the Layout before the Bridget upgrade is widely adopted, we could decide whether or not to include `InPageNewsletterSignup` or just render `ArticleBody` as before.

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx [hunk 1]
```diff
@@ -60,9 +104,38 @@ const NewsletterSignUpLayout: FC<Props> = ({ item, children }) => {
 					/>
 				</header>
 				<section css={contentRow}>
+					{!!newsletter && (
+						<div css={detailBlockStyles}>
+							<SvgNewsletter size="xsmall" />
+							<b>{newsletter.frequency}</b>
+							{/* TO DO - use regional focus, when on the MAPI type */}
+						</div>
+					)}
```
Comment (georgeblahblah) [inline apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:113]: we could use `maybeRender` here to help with this structure! `maybeRender` takes an `Option` as its first parameter, and then a function which returns React elements if that `Option` is a `some`:

```suggestion
					{maybeRender(item.promotedNewsletter, (newsletter) => (
						<div css={detailBlockStyles}>
							<SvgNewsletter size="xsmall" />
							<b>{newsletter.frequency}</b>
							{/* TO DO - use regional focus, when on the MAPI type */}
						</div>
					))}
```

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx [hunk 2]
```diff
@@ -60,9 +104,38 @@ const NewsletterSignUpLayout: FC<Props> = ({ item, children }) => {
...
+						<InPageNewsletterSignup
+							newsletter={newsletter}
+							format={getFormat(item)}
+							fallbackContent={
+								<ArticleBody format={item}>
+									{children}
+								</ArticleBody>
+							}
+						/>
+					)}
```
Comment (georgeblahblah) [inline apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:138]: we could use `maybeRender` here too

```suggestion
					{maybeRender(item.promotedNewsletter, (newsletter) => (
						<>
							<div css={frequencyBlockStyles}>
								<SvgClock size="xsmall" />

								<span>
									You&apos;ll receive this newsletter
									<b> {newsletter.frequency}</b>
								</span>
							</div>

							<InPageNewsletterSignup
								newsletter={newsletter}
								format={getFormat(item)}
								fallbackContent={
									<ArticleBody format={item}>
										{children}
									</ArticleBody>
								}
							/>
						</>
					))}
```

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx [hunk 3]
```diff
@@ -49,6 +89,10 @@ interface Props {
 
 const NewsletterSignUpLayout: FC<Props> = ({ item, children }) => {
 	const format = getFormat(item);
+	const newsletter =
+		item.promotedNewsletter.kind === OptionKind.Some
+			? item.promotedNewsletter.value
+			: undefined;
```
Comment (georgeblahblah) [inline apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:95]: if we use the `maybeRender` suggestions below

```suggestion
```

File: apps-rendering/src/components/InPageNewsletterSignup/NotSupportedMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import { InlineError } from '@guardian/source-react-components';
+import type { FC } from 'react';
+
+// ----- Component ----- //
+
+const NotSupportedMessage: FC = () => {
+	return (
+		<InlineError>
+			Your version of the app does not support newsletter sign-ups.
```
Comment (georgeblahblah) [inline apps-rendering/src/components/InPageNewsletterSignup/NotSupportedMessage.tsx:11]: I wonder if we could improve this message for people seeing it. Some ideas I had, but probably require product input:

- mention that they can upgrade using their app store
- if they are unable to upgrade the app, mention they can subscribe to the newsletter using the website

**Review by georgeblahblah — APPROVED** [review]
Thanks @dblatcher, I agree that removing the `initiallyRender` prop has simplified the component, although the story now inherits some of this complexity. I think this is an acceptable trade-off, especially as we don't easily support component-driven interactivity.

I had a suggestion about using `maybeRender` to conditionally render something depending on whether a value is a `Some`. We use it a lot in AR, and I think it's a good fit here as well.

Additionally I had some thoughts about the error message that users might see if their app is not the correct version.

**Comment by dblatcher** [comment]
> I had a suggestion about using `maybeRender` to conditionally render something depending on whether a value is a `Some`. We use it a lot in AR, and I think it's a good fit here as well.

Thanks - I'll give that a try and push the updates.

**Pull request merged**